library/src/main/java/org/apache/apex/malhar/lib/wal/FSWindowDataManager.java

```diff
@@ -51,291 +66,607 @@
 public class FSWindowDataManager implements WindowDataManager
```

Comment:
It may be useful to add a few lines WRT dynamic partitioning support and the relationship to checkpointing vs. commit per window.